Add glob support to podman run/create --mount (#19309)
containers/podman: docs/source/markdown/options/volume.md

Create a bind mount. If `-v /HOST-DIR:/CONTAINER-DIR` is specified, Podman
bind mounts `/HOST-DIR` from the host into `/CONTAINER-DIR` in the Podman
container. When specifying a host path, the host path must exist. Podman will
not create the host path and will return an error. File globbing is supported,

> --volume /HOST-DIR/foo\* as long as destination paths are not specified.

Reviewer: I feel like this needs its own paragraph, it's sufficiently different from the rest (no destination supported) that it feels like it needs to be distinct.

Similarly, `-v SOURCE-VOLUME:/CONTAINER-DIR` mounts the named volume from the
host into the container. If no such named volume exists,
Podman creates one. If no source is given, the volume is created
as an anonymously named volume with a randomly generated name, and is
removed when the <<container|pod>> is removed via the `--rm` flag or

> the `podman rm --volumes` command. If the file path glob matches no sources, it will also be treated as an anonymous named volume.

Reviewer: Does this make sense? Honestly, I'd rather we just mount nothing in this case.

Author: The problem I have is this would be a breaking change.  In current, Poman
``` 
$ podman run -v /tmp/foo\* alpine ls -l /tmp
total 0
drwxr-xr-x    2 root     root             6 Jul 21 18:36 foo*
```

Reviewer: Still seems very confusing. I'll think about this over the weekend, but mounting a named volume in this case is absolutely not what I would expect.

Author: Well there is no sure way to figure this out.  We could return an error if the user specifies a path that contains any 
"*", "?", "[" characters and the filepath.Glob did not match?  But this would prevent users from using volumes with these fileds in their name for a named volume.

Create a bind mount. If `-v /HOST-DIR:/CONTAINER-DIR` is specified, Podman
bind mounts `/HOST-DIR` from the host into `/CONTAINER-DIR` in the Podman
container. When specifying a host path, the host path must exist. Podman will
not create the host path and will return an error.

> File globbing is supported, --volume /HOST-DIR/foo\*. Not destination paths are
> not allowed. Paths matching the globs, will be mounted on the identical
> destination path within the container.

Reviewer 2: ```suggestion
File globbing is supported for sources (e.g., --volume /HOST-DIR/foo\*). 
Paths matching the globs, will be mounted on the identical
destination path within the container.
```

Similarly, `-v SOURCE-VOLUME:/CONTAINER-DIR` mounts the named volume from the
host into the container. If no such named volume exists,
Podman creates one. If no source is given, the volume is created
as an anonymously named volume with a randomly generated name, and is
removed when the <<container|pod>> is removed via the `--rm` flag or
the `podman rm --volumes` command. When the file path glob matches no paths on

> the host, the file path flob is treated as an anonymous named volume.

Reviewer 2: ```suggestion
the host, the file path glob is treated as an anonymous named volume.
```

Not sure why.  I would expect Podman to error out when the glob doesn't match.  Could be a user error which will be hard to debug when an anonymous volume is created instead.
